File: cpp/src/IceWS/Instance.cpp
// **********************************************************************
//
// Copyright (c) 2003-2014 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <IceWS/Instance.h>
#include <IceWS/EndpointI.h>
#include <Ice/ProtocolPluginFacade.h>

using namespace std;
using namespace Ice;
using namespace IceInternal;
using namespace IceWS;

IceUtil::Shared* IceWS::upCast(IceWS::Instance* p) { return p; }

IceWS::Instance::Instance(const CommunicatorPtr& communicator, Short type, const string& protocol) :
    ProtocolInstance(communicator, type, protocol)
{
}

IceWS::Instance::~Instance()
{
}

File: java/src/IceBT/src/main/java/com/zeroc/IceBT/Instance.java
// **********************************************************************
//
// Copyright (c) 2003-2017 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

package com.zeroc.IceBT;

import com.zeroc.Ice.Communicator;
import com.zeroc.Ice.PluginInitializationException;

import android.bluetooth.BluetoothAdapter;

class Instance extends com.zeroc.IceInternal.ProtocolInstance
{
    Instance(Communicator communicator, short type, String protocol)
    {
        super(communicator, type, protocol, type == com.zeroc.Ice.BTSEndpointType.value);

        _communicator = communicator;

        _bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
        if(_bluetoothAdapter == null)
        {
            throw new PluginInitializationException("bluetooth adapter not available");
        }
        else if(!_bluetoothAdapter.isEnabled())
        {
            throw new PluginInitializationException("bluetooth is not enabled");
        }
    }

    void destroy()
    {
        _communicator = null;
    }

    Communicator communicator()
    {
        return _communicator;
    }

    BluetoothAdapter bluetoothAdapter()
    {
        return _bluetoothAdapter;
    }

    private Communicator _communicator;
    private BluetoothAdapter _bluetoothAdapter;
}
